src/node.cc

```diff
@@ -2498,36 +2502,96 @@ struct node_module* get_linked_module(const char* name) {
   return mp;
 }
 
-// DLOpen is process.dlopen(module, filename).
+struct DLib {
+  std::string filename_;
+  std::string errmsg_;
+  void* handle_;
+  int flags_;
+
+  void* Handle() {
+    return handle_;
+  }
+
+  const char* GetError() {
+    return errmsg_.c_str();
+  }
```

Comment:
Since these methods are simple wrappers and only used once, I'd simply inline them at the call site.

Reply:
You mean like this `Local<String> errmsg = OneByteString(env->isolate(), dlib.errmsg_.c_str());` @bnoordhuis ?